plugins/Goals/Controller.php

```diff
@@ -335,24 +334,36 @@ protected function getTopDimensions($idGoal)
 
         $topDimensions = array();
         foreach ($topDimensionsToLoad as $dimensionName => $apiMethod) {
+            if ($apiMethod == 'Actions.getEntryPageUrls') {
+                $columnNbConversions = 'goal_' . $idGoal . '_nb_conversions_entry';
+                $columnConversionRate = 'goal_' . $idGoal . '_nb_conversions_entry_rate';
+                $idGoalToProcess = AddColumnsProcessedMetricsGoal::GOALS_ENTRY_PAGES;
+            } else {
+                $columnNbConversions = 'goal_' . $idGoal . '_nb_conversions';
+                $columnConversionRate = 'goal_' . $idGoal . '_conversion_rate';
+                $idGoalToProcess = AddColumnsProcessedMetricsGoal::GOALS_FULL_TABLE;
+            }
+
             $request = new Request("method=$apiMethod
                                    &format=original
+                                   &format_metrics=0
                                    &filter_update_columns_when_show_all_goals=1
-                                   &idGoal=" . AddColumnsProcessedMetricsGoal::GOALS_FULL_TABLE . "
+                                   &idGoal=$idGoalToProcess
                                    &filter_sort_order=desc
-                                   &filter_sort_column=$columnNbConversions" .
+                                   &filter_sort_column=$columnNbConversions
+                                   &showColumns=label,$columnNbConversions,$columnConversionRate" .
                 // select a couple more in case some are not valid (ie. conversions==0 or they are "Keyword not defined")
                 "&filter_limit=" . (self::COUNT_TOP_ROWS_TO_DISPLAY + 2));
-            $datatable = $request->process();
-            $topDimension = array();
-            $count = 0;
 
             if ($apiMethod == 'Actions.getEntryPageUrls') {
-                $columnConversionRate = 'goal_' . $idGoal . '_nb_conversions_entry_rate';
-            } else {
-                $columnConversionRate = 'goal_' . $idGoal . '_conversion_rate';
+                $request .= '&flat=1';
```

Comment:
it seems like the `$request` is an object, maybe need to update the whole thing like below.

```php
       $requestString = "method=$apiMethod
                                   &format=original
                                   &format_metrics=0
                                   &filter_update_columns_when_show_all_goals=1
                                   &idGoal=$idGoalToProcess
                                   &filter_sort_order=desc
                                   &filter_sort_column=$columnNbConversions
                                   &showColumns=label,$columnNbConversions,$columnConversionRate" .
                // select a couple more in case some are not valid (ie. conversions==0 or they are "Keyword not defined")
                "&filter_limit=" . (self::COUNT_TOP_ROWS_TO_DISPLAY + 2);

            if ($apiMethod == 'Actions.getEntryPageUrls') {
                $requestString .= '&flat=1';
            }

            $request = new Request($requestString);

```

Reply:
Ah sure. Should be fixed.